### DISCUSSION TITLE
A question about sensitivity analyses!

### DISCUSSION THREAD

**Discussion by Harris728** [Open-Systems-Pharmacology/Forum#1223] [Questions & Problems]
Dear all,

Could you tell me how to determine the values of "number of steps" and "variation range"? When I set "number of steps" at 9, and "variation range" at 10, I found the results were significant different from that of the default settings, and it would exaggerate sensitivity.

For example, when I use default settings, n=2 and a=0.1, the sensitivity values of fu and pKa were -0.91 and -0.82, respectively. When I set "number of steps" at 9, and "variation range" at 10, the sensitivity values of fu and pKa were -3.36 and -0.1, respectively. 

I want to know which one is correct and how to determine the values of "number of steps" and "variation range".

Thanks.

**Comment by StephanSchaller** [comment]
you will have to think about how parameters affect your model.
The pKa might be sensitive (also depending on other drug properties) if it is set at physiologic relevant values (pH 2 or 5 to 7.5 or maybe 8). also, it will have no real meaning > 14  etc. So starting at e.g. 5 and increasing it 10-fold ("variation range" at 10) will not make much sense.

For a local sensitivity analysis, it is best to keep the "variation range" at very small values (I would even say 0.1 could be considered a sufficiently large). 
Setting the "variation range" to larger values (and 10 might be too extreme (see example above)) might help to identify non-linearities, if it returns a different value...

**Reply by Harris728** [reply]
Thank you so much for your response, Stephan. I don't know if I have got what you said. Do you mean that the "variation range" is used to reflect the fold of variation of the parameter? If I set it too extreme, the parameter will fall into a meaningless range. So I'd better to keep it at a small value. By the way, could you tell me what the role of "number of steps" is?

I have read the documentation of PK-sim, but I can't understand the description of this section. I really thank you for your explanation.

**Reply by msevestre** [reply]
see the documentation 
https://docs.open-systems-pharmacology.org/shared-tools-and-example-workflows/sensitivity-analysis

![image](https://user-images.githubusercontent.com/1041237/172424847-eff96f90-a1ea-4352-83f3-a8af86a1631c.png)

Let's assume n =2 and a = 0.1
we will generate the following variation factor, by having the parameter k going between 1 and 2 (because n=2)

k =1

Variation1: 1 + 0.1 * 1/2 = 1+0.05 = **1.05**
Variation2: 1/(1 + 0.1 * 1/2) = **1/(1.05)**

k=2
Variation3: 1 + 0.1 * 2/2 = 1+0.1 = **1.1**
Variation4: 1/(1 + 0.1 * 2/2) = **1/(1.1)**

Then those values are multiplied by the value of the parameter to create a variation around the parameter. Does this make sense?

So if your parameter value is 8, you will get 4 values: 8 x1.05, 8x1.1, etc..

**Reply by Harris728** [reply]
I see. Thank you so much.
